unit-testing/src/mock-function/lib/insert.spec.ts

```diff
@@ -0,0 +1,118 @@
+// insertDocument.test.ts
+import { Container } from '../data/connect-to-cosmos';
+import { createTestInputAndResult } from '../data/fake-data';
+import {
+  DbDocument,
+  DbError,
+  isDbError,
+  isVerificationErrors,
+  RawInput,
+} from '../data/model-schema';
+import { inputVerified } from '../data/verify';
+import { insertDocument } from './insert';
+
+// Mock app dependencies for Cosmos DB setup
+jest.mock('../data/connect-to-cosmos', () => ({
```

Comment:
it would be helpful to call out that for larger modules where you only want to mock part of the implementation (and not the whole thing) you could use partial mocks https://jestjs.io/docs/mock-functions#mocking-partials

In this case it's fine since the module is only 2 functions - so I don't know if it's worth actually adding the code from the linked doc - but maybe calling that out explicitly somewhere would be helpful (what should I do for larger modules)

Reply:
I'll call it out in the doc and leave the code as is.